appserver/common/glassfish-naming/src/main/java/com/sun/enterprise/naming/util/NamingUtilsImpl.java

```diff
@@ -107,42 +106,41 @@ public NamingObjectFactory createDelegatingNamingObjectFactory(String name,
         NamingObjectFactory delegate, boolean cacheResult) {
         return new DelegatingNamingObjectFactory(name, delegate, cacheResult);
     }
-    
+
     @Override
     public Object makeCopyOfObject(Object obj) {
         if ( !(obj instanceof Context) && (obj instanceof Serializable) ) {
             if(logger.isLoggable(Level.FINE)) {
                 logger.log(Level.FINE, "** makeCopyOfObject:: " + obj);
             }
-            
             try {
                 // first serialize the object
-                ByteArrayOutputStream bos = new ByteArrayOutputStream();
-                ObjectOutputStream oos = getFactory().createObjectOutputStream(bos);
-                oos.writeObject(obj);
-                oos.flush();
-                byte[] data = bos.toByteArray();
-                oos.close();
-                bos.close();
-
+                byte[] data = serialize(obj);
                 // now deserialize it
-                ByteArrayInputStream bis = new ByteArrayInputStream(data);
-                final ObjectInputStream ois = getFactory().createObjectInputStream(bis);
-                obj = AccessController.doPrivileged(new PrivilegedExceptionAction() {
-                    @Override
-                    public Object run() throws IOException, ClassNotFoundException {
-                        return ois.readObject();
-                    }
-                });
-                return obj;
+                return deserialize(data);
             } catch (Exception ex) {
                 logger.log(Level.SEVERE, EXCEPTION_COPY_MUTABLE, ex);
-                RuntimeException re = new RuntimeException("Cant copy Serializable object:", ex);
-                throw re;
+                throw new RuntimeException("Cant copy Serializable object:", ex);
             }
         } else {
             // XXX no copy ?
             return obj;
         }
     }
+
+    private Object deserialize(byte[] data) throws IOException, java.security.PrivilegedActionException {
+        try(final ObjectInputStream ois =
+                getFactory().createObjectInputStream(new ByteArrayInputStream(data))){
+            return AccessController.doPrivileged((PrivilegedExceptionAction) () -> ois.readObject());
+        }
+    }
+
+    private byte[] serialize(Object obj) throws IOException {
+        try(ByteArrayOutputStream bos = new ByteArrayOutputStream();
+            ObjectOutputStream oos = getFactory().createObjectOutputStream(bos)) {
```

Comment:
Formatting is a little off here, makes this a bit hard to read as this line looks like it belongs with the ones below.

Reply:
This is actually 2 Closables in one try-with-resources block. To have them both automatically closed. I have made now 2 try blocks to make clearer.